src/ImageSharp/Processing/Processors/Convolution/MedianRowOperation{TPixel}.cs

```diff
@@ -0,0 +1,171 @@
+            this.configuration = configuration;
+            this.targetPixels = targetPixels;
+            this.sourcePixels = sourcePixels;
+            this.map = map;
+            this.kernelSize = kernelSize;
+            this.preserveAlpha = preserveAlpha;
+            int kernelCount = this.kernelSize * this.kernelSize;
+            this.yChannelStart = kernelCount;
+            this.zChannelStart = this.yChannelStart + kernelCount;
+            this.wChannelStart = this.zChannelStart + kernelCount;
+        }
+
+        public void Invoke(int y, Span<Vector4> span)
+        {
+            // Span has kernelSize^2 followed by bound width.
+            int boundsX = this.bounds.X;
+            int boundsWidth = this.bounds.Width;
+            int kernelCount = this.kernelSize * this.kernelSize;
+            Span<Vector4> kernelBuffer = span.Slice(0, kernelCount);
+            Span<Vector4> channelVectorBuffer = span.Slice(kernelCount, kernelCount);
+            Span<Vector4> targetBuffer = span.Slice(kernelCount << 1, boundsWidth);
+
+            // Stack 4 channels of floats in the space of Vector4's.
+            Span<float> channelBuffer = MemoryMarshal.Cast<Vector4, float>(channelVectorBuffer);
+            Span<float> xChannel = channelBuffer.Slice(0, kernelCount);
+            Span<float> yChannel = channelBuffer.Slice(this.yChannelStart, kernelCount);
+            Span<float> zChannel = channelBuffer.Slice(this.zChannelStart, kernelCount);
+
+            DenseMatrix<Vector4> kernel = new DenseMatrix<Vector4>(this.kernelSize, this.kernelSize, kernelBuffer);
+
+            int row = y - this.bounds.Y;
+            MedianConvolutionState state = new MedianConvolutionState(in kernel, this.map);
+            ref int sampleRowBase = ref state.GetSampleRow(row);
+            ref Vector4 targetBase = ref MemoryMarshal.GetReference(targetBuffer);
+
+            if (this.preserveAlpha)
+            {
+                for (int x = 0; x < boundsWidth; x++)
+                {
+                    int index = 0;
+                    ref int sampleColumnBase = ref state.GetSampleColumn(x);
+                    ref Vector4 target = ref Unsafe.Add(ref targetBase, x);
+                    for (int kY = 0; kY < state.Kernel.Rows; kY++)
+                    {
+                        int currentYIndex = Unsafe.Add(ref sampleRowBase, kY);
+                        Span<TPixel> sourceRow = this.sourcePixels.DangerousGetRowSpan(currentYIndex).Slice(boundsX, boundsWidth);
+                        ref TPixel sourceRowBase = ref MemoryMarshal.GetReference(sourceRow);
+                        for (int kX = 0; kX < state.Kernel.Columns; kX++)
+                        {
+                            int currentXIndex = Unsafe.Add(ref sampleColumnBase, kX) - boundsX;
+                            TPixel pixel = Unsafe.Add(ref sourceRowBase, currentXIndex);
+                            state.Kernel.SetValue(index, pixel.ToVector4());
```

Comment:
Now you can set the value straight from our source `Vector4` buffer

Reply:
Done, while picking the correct source row.